### PR TITLE
Feature/39479 Added detection for ie9/ie10 and flag as unsupported browsers

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
Modified in both login and connectivity check component

### CODE REVIEW

File: pupil-spa/src/app/connectivity-check/connectivity-check.component.html [hunk 1]
```diff
@@ -10,5 +10,14 @@
       </div>
     </div>
   </div>
+  <div *ngIf="isUnsupportedBrowser === true" class="grid-row">
+    <header>
+      <h1 class="heading-xlarge">This broswer is not supported</h1>
```
Comment (jon-shipley) [inline pupil-spa/src/app/connectivity-check/connectivity-check.component.html]:
```suggestion
      <h1 class="heading-xlarge">This browser is not supported</h1>
```

File: pupil-spa/src/app/connectivity-check/connectivity-check.component.html [hunk 2]
```diff
@@ -10,5 +10,14 @@
...
+    </header>
+    <p class="lede" tabindex="0">
+      You can view additional information on supported browsers
+      in the MTC technical guidance on GOV.UK: <a href="https://www.gov.uk/government/collections/multiplication-tables-check">https://www.gov.uk/government/collections/multiplication-tables-check</a>
```
Comment (jon-shipley) [inline pupil-spa/src/app/connectivity-check/connectivity-check.component.html]: Whilst 'technical guidance' may be the approved copy, the guidance appears to be in Annex  A of the 'System Guidance' and if the reader does manage to find it, it says:

> 'The MTC has been designed to support all common browsers. However, please note
> certain optional features, such as the question reader, will not work on Internet Explorer
> or Edge due to the technical limitations of these browsers.'

I'd suggest changing 'MTC technical guidance' to 'MTC system guidance' which is a real thing.

File: pupil-spa/src/app/login/login.component.html [hunk 3]
```diff
@@ -57,6 +57,15 @@ <h1 class="heading-xlarge">Sign in</h1>
         </form>
       </div>
     </div>
+    <div *ngIf="isUnsupportedBrowser === true" class="grid-row">
+      <header>
+        <h1 class="heading-xlarge">This broswer is not supported</h1>
```
Comment (jon-shipley) [inline pupil-spa/src/app/login/login.component.html]:
```suggestion
        <h1 class="heading-xlarge">This browser is not supported</h1>
```

File: pupil-spa/src/app/login/login.component.html [hunk 4]
```diff
@@ -57,6 +57,15 @@ <h1 class="heading-xlarge">Sign in</h1>
...
+      </header>
+      <p class="lede" tabindex="0">
+        You can view additional information on supported browsers
+        in the MTC technical guidance on GOV.UK: <a href="https://www.gov.uk/government/collections/multiplication-tables-check">https://www.gov.uk/government/collections/multiplication-tables-check</a>
```
Comment (jon-shipley) [inline pupil-spa/src/app/login/login.component.html]: Same comment as before about 'system guidance'

File: pupil-spa/src/app/services/device/device.service.spec.ts [hunk 5]
```diff
@@ -31,4 +41,21 @@ describe('DeviceService', () => {
     const deviceInfo = storageService.getDeviceData();
     expect(deviceInfo).toBeTruthy();
   });
+
+  describe('isUnsupportedBrowser', () => {
+    beforeEach(() => {
```
Comment (jon-shipley) [inline pupil-spa/src/app/services/device/device.service.spec.ts]: Needs negative tests too: last two version of Chrome, Firefox, Safari, Edge

File: pupil-spa/src/app/connectivity-check/connectivity-check.component.html (hunk 2 above)
Comment (jon-shipley) [reply pupil-spa/src/app/connectivity-check/connectivity-check.component.html]: Also, Edge does support the Speech Synthesis API and the latest Chromium builds will be fine.
